src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/TypeName.cs

```diff
@@ -401,6 +431,22 @@ public static bool TryParse(ReadOnlySpan<char> typeName, [NotNullWhen(true)] out
             return result is not null;
         }
 
+        /// <summary>
+        /// Converts any escaped characters in the input type name or namespace.
+        /// </summary>
+        /// <param name="name">The input string containing the name to convert.</param>
+        /// <returns>A string of characters with any escaped characters converted to their unescaped form.</returns>
+        /// <remarks>The unescaped string can be used for looking up the type name or namespace in metadata.</remarks>
```

Comment:
Also mention that the API does not validate that the escape sequences are only used for characters that require escaping?

Reply:
Updated.